llms/hf_llm/convert.py

```diff
@@ -0,0 +1,175 @@
+# Copyright © 2023 Apple Inc.
+
+import argparse
+import copy
+import glob
+import json
+from pathlib import Path
+
+import mlx.core as mx
+import mlx.nn as nn
+import transformers
+from huggingface_hub import snapshot_download
+from mlx.utils import tree_flatten, tree_map
+from models import Model, ModelArgs
+
+
+def fetch_from_hub(hf_path: str):
+    model_path = snapshot_download(
+        repo_id=hf_path,
+        allow_patterns=["*.json", "*.safetensors", "tokenizer.model"],
+    )
+    weight_files = glob.glob(f"{model_path}/*.safetensors")
+    if len(weight_files) == 0:
+        raise FileNotFoundError("No safetensors found in {}".format(model_path))
+
+    weights = {}
+    for wf in weight_files:
+        weights.update(mx.load(wf).items())
+
+    config = transformers.AutoConfig.from_pretrained(hf_path)
+    tokenizer = transformers.AutoTokenizer.from_pretrained(
+        hf_path,
+    )
+    return weights, config.to_dict(), tokenizer
+
+
+def quantize(weights, config, args):
+    quantized_config = copy.deepcopy(config)
+
+    # Load the model:
+    model = Model(ModelArgs.from_dict(config))
+    weights = tree_map(mx.array, weights)
```

Comment:
When would the weights not be an `mx.array`? Loaded with numpy or sth?

Reply:
Oversight, good catch.